src/Compiler/Interactive/fsi.fs

```diff
@@ -3631,48 +3631,52 @@ type FsiEvaluationSession (fsi: FsiEvaluationSessionHostConfig, argv:string[], i
         // Explanation: When the user of the FsiInteractiveSession object calls this method, the
         // code is parsed, checked and evaluated on the calling thread. This means EvalExpression
         // is not safe to call concurrently.
+        let scrptFileName = defaultArg scriptFileName dummyScriptFileName
         let ctok = AssumeCompilationThreadWithoutEvidence()
 
         let errorOptions = TcConfig.Create(tcConfigB,validate = false).diagnosticsOptions
         let diagnosticsLogger = CompilationDiagnosticLogger("EvalInteraction", errorOptions, eagerFormat)
-        fsiInteractionProcessor.EvalExpression(ctok, code, dummyScriptFileName, diagnosticsLogger)
-        |> commitResultNonThrowing errorOptions dummyScriptFileName diagnosticsLogger
+        fsiInteractionProcessor.EvalExpression(ctok, code, scrptFileName, diagnosticsLogger)
+        |> commitResultNonThrowing errorOptions scrptFileName diagnosticsLogger
 
-    member _.EvalInteraction(code, ?cancellationToken) : unit =
+    member _.EvalInteraction(code, ?cancellationToken, ?scriptFileName:string) : unit =
         // Explanation: When the user of the FsiInteractiveSession object calls this method, the
         // code is parsed, checked and evaluated on the calling thread. This means EvalExpression
         // is not safe to call concurrently.
         let ctok = AssumeCompilationThreadWithoutEvidence()
         let cancellationToken = defaultArg cancellationToken CancellationToken.None
-        fsiInteractionProcessor.EvalInteraction(ctok, code, dummyScriptFileName, diagnosticsLogger, cancellationToken)
+        let scrptFileName = defaultArg scriptFileName dummyScriptFileName
+        fsiInteractionProcessor.EvalInteraction(ctok, code, scrptFileName, diagnosticsLogger, cancellationToken)
         |> commitResult
         |> ignore
 
-    member _.EvalInteractionNonThrowing(code, ?cancellationToken) =
+    member _.EvalInteractionNonThrowing(code, ?cancellationToken, ?scriptFileName:string) =
         // Explanation: When the user of the FsiInteractiveSession object calls this method, the
         // code is parsed, checked and evaluated on the calling thread. This means EvalExpression
         // is not safe to call concurrently.
         let ctok = AssumeCompilationThreadWithoutEvidence()
         let cancellationToken = defaultArg cancellationToken CancellationToken.None
+        let scrptFileName = defaultArg scriptFileName dummyScriptFileName
 
         let errorOptions = TcConfig.Create(tcConfigB,validate = false).diagnosticsOptions
         let diagnosticsLogger = CompilationDiagnosticLogger("EvalInteraction", errorOptions, eagerFormat)
-        fsiInteractionProcessor.EvalInteraction(ctok, code, dummyScriptFileName, diagnosticsLogger, cancellationToken)
-        |> commitResultNonThrowing errorOptions "input.fsx" diagnosticsLogger
+        fsiInteractionProcessor.EvalInteraction(ctok, code, scrptFileName, diagnosticsLogger, cancellationToken)
```

Comment:
spelling : scrptFileName

Reply:
Thanks, this local value will be obsolete when moving to overloads.